src/functions.cpp

```diff
@@ -1,6 +1,6 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
-#include <algorithm>
 #include <iostream>
+#include <algorithm>
+#include "functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
 #include <limits>
 
 void print(std::vector<int> const & numbers) {
```

Comment:
Is this for loop work with int i? I'm strongly recommend to use size_t instead of int.